### DISCUSSION TITLE
Вывод данные на печать

### DISCUSSION THREAD

**Discussion by Re-Requeste** [totumonline/totum-ru-issues-and-discussions#404] [Как это сделать?] [answered]
**Доброго времени суток.**

Пытаюсь реализовать печать данных из строчной части, но в строчной части существует _данные которые я вытащил из циклов_.

В данные входят : список позиции, количество и умноженная сумма.
Возможно ли вытащить данные ? 

Если да, есть ли возможность разделить их по строчкам в форме html (td = > Наименование td = > колво => сумма)

**Скриншот того что буду печатать :**

![Безымянный](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/137899742/6f57a3b4-94b3-49b8-bf86-8d1861238d06)

**Примерно так будет  выглядит печать:**

![Безымянный 2](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/137899742/882b5d58-7d3d-4462-baf3-642901cc4ee2)

**Если коротко описать, как из json вытащить данные и передать эти значения на печать. Спасибо ^^**

**Comment by Re-Requeste** [comment]
Учитывая что у меня строчная часть повторяется, и в одном из них лежат данные которые я вытаскиваю в элемент шаблона : shet_m, и дальше просто забираю те строки которые мне нужны.

Попытался через listSection обратится за данными в ячейку, но к сожалению шаблон считает что он простой Row и берет только 1 значение, ну это и понятно раз я использую select, но если я использую selectList то там мы ошибку получаем что лист в листе не получится создать.. Вообщем, пойду дальше голову ломать :)
```
###code ###
innerrowlist:  rowCreate(field: "template" = "shet_m"; field: "data" = $rl)
rl: rowListCreate(field: "pozicii_v_schete" = $fieldList[[pozicii_v_schete]]; field: "zakaz" = $fieldList[[zakaz]]; field: "testing" = $listing)

listing: listSection(list: $selrow; item: "cena_za_ed")
	selrow: select(table: $#ntn; field: 'dannye_vnutri_scheta')
```

**Comment by erpic-ru** [comment]
Сходу можно придумать 2 решения:

1. Сделать и вывести нормальную простую таблицу в формате Заказчик - Клиент - Позиция - Кол-во - Сумма ...

2. С помощью listreplace, str, listtrain и strreplace собрать нужный html со всеми объединениями ячеек и передать его переменной в печатную форму. Я бы даже сделал техническое поле под каждую строку, где собирал бы этот код с тегами текстом, а затем вывел бы его списком через элемент печатной формы

**Comment by poluninatm** [comment, marked as answer]
Функции linkToPrint и textByTemplate могут обрабатывать вложенные шаблоны. Вариант, как это можно собрать через textByTemplate:

<img width="875" alt="Снимок экрана 2023-07-05 в 08 52 24" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/2241864/2b0c0d19-f95e-4f8d-84f1-e9e9362fc373">

<img width="1190" alt="
Снимок экрана 2023-07-05 в 08 51 37" src="https://github.com/totumonline/totum-ru-issues-and-discussions/assets/2241864/43f6c6d4-4fa0-4a5e-8bc1-5d04879e3cb3">
